packages/vscode-extension/src/devices/preview.ts

```diff
@@ -102,39 +159,28 @@ export class Preview implements Disposable {
     this.subprocess?.stdin?.write("pointer show false\n");
   }
 
+  public startRecording() {
+    this.sendCommandOrThrow(`video recording start -m -b 50\n`); // 50MB buffer for in-memory video
+  }
+
+  public stopRecording() {
+    this.sendCommandOrThrow(`video recording stop\n`);
+  }
+
+  public captureRecording() {
+    return this.handleVideoRecordingPromise("recording");
+  }
+
   public startReplays() {
-    const stdin = this.subprocess?.stdin;
-    if (!stdin) {
-      throw new Error("sim-server process not available");
-    }
-    stdin.write(`video replay start -m -b 50\n`); // 50MB buffer for in-memory video
+    this.sendCommandOrThrow(`video replay start -m -b 50\n`); // 50MB buffer for in-memory video
   }
 
   public stopReplays() {
-    const stdin = this.subprocess?.stdin;
-    if (!stdin) {
-      throw new Error("sim-server process not available");
-    }
-    stdin.write(`video replay stop\n`);
+    this.sendCommandOrThrow(`video replay stop\n`);
   }
 
   public captureReplay() {
-    const stdin = this.subprocess?.stdin;
-    if (!stdin) {
-      throw new Error("sim-server process not available");
-    }
-    let resolvePromise: (value: RecordingData) => void;
-    let rejectPromise: (reason?: any) => void;
-    const promise = new Promise<RecordingData>((resolve, reject) => {
-      resolvePromise = resolve;
-      rejectPromise = reject;
-    });
-    if (this.lastReplayPromise) {
-      promise.then(this.lastReplayPromise.resolve, this.lastReplayPromise.reject);
-    }
-    this.lastReplayPromise = { resolve: resolvePromise!, reject: rejectPromise! };
-    stdin.write(`video replay save\n`);
-    return promise;
+    return this.handleVideoRecordingPromise("replay");
```

Comment:
`handleVideoRecordingPromise` is a weird name. There's no promise at this point, so what is this method "handling"??

I'd just name it `saveVideo` or `saveVideoWithID`